program-runtime/src/invoke_context.rs

```diff
@@ -369,8 +369,13 @@ impl<'a> InvokeContext<'a> {
             }
 
             self.pre_accounts = Vec::with_capacity(instruction_accounts.len());
-            let mut work = |_index_in_instruction: usize,
-                            instruction_account: &InstructionAccount| {
+
+            for (index_in_instruction, instruction_account) in
```

Comment:
Having visit each account once was a clear way to indicating what was happening here, this is far less obvious

Reply:
I think `visit_each_account_once()` was actually not that descriptive of what it was doing after all. It should have been called `visit_each_unique_account_once()` or `iterate_skipping_duplicate_accounts()`.

Reply:
Or just `visit_each_unique_account`